### PR TITLE
Preparations for .NET Core 3

### PR DESCRIPTION
**Pull request by Shtong**
This PR contains changes to prepare the library to work with .NET Core 3.
- The website has been updated to use ASP.NET Core 3.0 instead of ASP.NET Core 2.2
- The FIDO2 library now also targets `netcoreapp3`

I haven't changed the library code, and everything seems to be in order.

Note: I plan to update this branch again once the final versions of the 3.0 dependencies are released. This PR can be reviewed, as there won't be much changes needed to switch to the final dependencies, but please wait until I've done the update before you merge :)

### CODE REVIEW

**Comment by abergs** [comment]
We will keep this open until core 3 is released and build succeeds. Thank you for your work!

**Comment by Shtong** [comment]
I have updated the web project to use the final version of the 3.0.0 dependencies. Now someone needs to update the build process to use the 3.0 core SDK for building instead of 2.2.101.

File: Directory.Build.props [hunk 1]
```diff
@@ -15,7 +15,7 @@
   <!-- GLOBALLY USABLE VARIABLES -->
   <PropertyGroup>
     <!-- Establish a preset but make it an active choice for each project. -->
-    <SupportedNonMetaTargetFrameWorks>netcoreapp2.2;netcoreapp3.0</SupportedNonMetaTargetFrameWorks>
+    <SupportedNonMetaTargetFrameWorks>netcoreapp3.0</SupportedNonMetaTargetFrameWorks>
```
Comment (Shtong) [inline Directory.Build.props]: The reason why I chose to remove the use of `$(SupportedNonMetaTargetFrameWorks)` in `Demo.csproj` is that it doesn't make sense for a website to target two runtimes at the same time. However, I think it makes sense that `$(SupportedNonMetaTargetFrameWorks)` does contain all the runtimes the library is supposed to support. This list might be useful in several places, while the runtime used by the demo website isn't.
A nice side effect is that the unit tests will run twice (once for every runtime), which will make sure that the library stays compatible with `netcoreapp2.2` and `netcoreapp3.0`.
I think we should revert the changes in `Demo.csproj` and `Directory.Build.props` in order to keep unit testing on both runtimes.

Comment (aseigler) [reply Directory.Build.props]: I couldn't get the project to work on both runtimes on all platforms at the same time time.

Comment (Shtong) [reply Directory.Build.props]: Was it not working on your environment or on the CI build? I had no problem on my end.

Comment (aseigler) [reply Directory.Build.props]: It was in my environment. I will run through it again.

**Comment by abergs** [comment]
@Shtong Nice work. As you might already noticed I exported the build configuration YAML here: https://github.com/abergs/fido2-net-lib/issues/126#issuecomment-546883705 so that it's possible to change the build target for the branch

**Comment by Shtong** [comment]
Here's an updated version of this branch. The news are :
- I have updated `azure-pipelines.yml` to contain the current build process. However, I believe it can be improved; see the discussion in #126 
- I have replaced the deprecated `PackageLicenceUrl` project property, to use `<PackageLicenseExpression>MIT</PackageLicenseExpression>` instead. This would fix #135 
- I have messed up in my commits, and force pushed a set of changes without thinking that @aseigler made changes on the branch. So, some of these changes may have disappeared! Apologies. You may want to check that everything works before giving the green light.

**Comment by mackie1001** [comment]
I'm keen to upgrade our solution to 3.1 from 2.1 and am happy to lend a hand. What's the official plan re implementing support?

**Comment by Shtong** [comment]
The build results are gone now, but if I remember correctly the problem was that the build pipeline was configured to use the core 2.2 toolset, instead of 3.0/3.1. I was waiting for feedback on #126 to update the build pipeline but unfortunately there has not been any activity in the last few months...

**Comment by aseigler** [comment]
I've reached out to @abergs for assistance.

**Comment by abergs** [comment]
Quick status: I'm back and looking through everything.

**Comment by abergs** [comment]
Demo build is currently failing with these logs:

```
2020-03-02T22:18:33.4402031Z ##[section]Starting: dotnet build entire sln
2020-03-02T22:18:33.4523770Z ==============================================================================
2020-03-02T22:18:33.4524418Z Task         : .NET Core
2020-03-02T22:18:33.4524978Z Description  : Build, test, package, or publish a dotnet application, or run a custom dotnet command
2020-03-02T22:18:33.4525500Z Version      : 2.164.0
2020-03-02T22:18:33.4525869Z Author       : Microsoft Corporation
2020-03-02T22:18:33.4526412Z Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/build/dotnet-core-cli
2020-03-02T22:18:33.4526994Z ==============================================================================
2020-03-02T22:18:34.1012746Z [command]C:\windows\system32\chcp.com 65001
2020-03-02T22:18:34.1108030Z Active code page: 65001
2020-03-02T22:18:34.3306926Z [command]C:\hostedtoolcache\windows\dncs\2.2.101\x64\dotnet.exe publish d:\a\1\s\Demo\Demo.csproj --configuration Release --output d:\a\1\a\Demo --framework netcoreapp2.2
2020-03-02T22:18:34.6619460Z Microsoft (R) Build Engine version 16.4.0+e901037fe for .NET Core
2020-03-02T22:18:34.6620948Z Copyright (C) Microsoft Corporation. All rights reserved.
2020-03-02T22:18:34.6621755Z 
2020-03-02T22:18:36.3443578Z   Restore completed in 376.61 ms for d:\a\1\s\Src\Fido2.AspNet\Fido2.AspNet.csproj.
2020-03-02T22:18:36.3445173Z   Restore completed in 380.74 ms for d:\a\1\s\Demo\Demo.csproj.
2020-03-02T22:18:36.4131499Z   Restore completed in 12.8 ms for d:\a\1\s\Src\Fido2.Models\Fido2.Models.csproj.
2020-03-02T22:18:36.4134106Z   Restore completed in 29.38 ms for d:\a\1\s\Src\Fido2\Fido2.csproj.
2020-03-02T22:18:36.7369855Z   Restore completed in 330.2 ms for d:\a\1\s\Demo\Demo.csproj.
2020-03-02T22:18:37.5160682Z C:\Program Files\dotnet\sdk\3.1.101\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(234,5): error NETSDK1005: Assets file 'd:\a\1\s\Demo\obj\project.assets.json' doesn't have a target for '.NETCoreApp,Version=v2.2'. Ensure that restore has run and that you have included 'netcoreapp2.2' in the TargetFrameworks for your project. [d:\a\1\s\Demo\Demo.csproj]
2020-03-02T22:18:37.5595055Z ##[error]Error: The process 'C:\hostedtoolcache\windows\dncs\2.2.101\x64\dotnet.exe' failed with exit code 1
2020-03-02T22:18:37.5611841Z ##[warning]Info: Azure Pipelines hosted agents have been updated to contain .Net Core 3.x SDK/Runtime along with 2.2 & 2.1. Unless you have locked down a SDK version for your project(s), 3.x SDK might be picked up which might have breaking behavior as compared to previous versions. 
Some commonly encountered changes are: 
If you're using `Publish` command with -o or --Output argument, you will see that the output folder is now being created at root directory rather than Project File's directory. To learn about more such changes and troubleshoot, refer here: https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/build/dotnet-core-cli?view=azure-devops#troubleshooting
2020-03-02T22:18:37.5617486Z ##[error]Dotnet command failed with non-zero exit code on the following projects : d:\a\1\s\Demo\Demo.csproj
2020-03-02T22:18:37.5639356Z ##[section]Finishing: dotnet build entire sln

```

**Comment by abergs** [comment]
I ran a new build configuration with the piplinesfile and it worked out well:

image

**Comment by abergs** [comment]
Is there anything we need to review in regard to the comment from nov 2?

> I have messed up in my commits, and force pushed a set of changes without thinking that @aseigler made changes on the branch. So, some of these changes may have disappeared! Apologies. You may want to check that everything works before giving the green light.

Otherwise if you @Shtong don't have more edits you wish to make I will check out the branch, run it locally and make sure the behavior is as expected.

**Comment by Shtong** [comment]
I think I now have a good initial version.
Some closing notes for this PR:
- I took the opportunity to upgrade the microsoft packages to the latest 3.1 version
- There were two commits that should not have been in that branch (e5d06f2 and 
b3196d8), which I reverted today. Sorry for the confusion
- In the pipeline file, all I do to build the library is invoke the already existing build script (`buildRelease.ps1`). I may be a good idea in the future to replace that PS script with tasks the the pipeline file.
- The pipeline only consists on one file, in which I have put the three old build processes in separated jobs. That being said, it would be easy to split that file in two if pipeline jobs are not enough of a separation (as discussed in #126 )

The PR is now ready to merge if you guys do not suggest more changes.

**Comment by abergs** [comment]
@Shtong I think it still needs to be rebased before merge (GitHub UI says so)

**Comment by Shtong** [comment]
That's weird, my GitHub interface says there are no conflicts, and the git shell says my local branch is up to date compared to you master branch.

image

**Comment by abergs** [comment]
Merged in #145

**Pull request closed without merging**